db.go

```diff
@@ -112,9 +112,13 @@ type DB struct {
 	donec    chan struct{}
 	stopc    chan struct{}
 
-	// cmtx is used to control compactions and deletions.
-	cmtx               sync.Mutex
-	compactionsEnabled bool
+	// cmtx ensures that compactions and deletions don't run simultaneously.
+	cmtx sync.Mutex
+
+	// autoCompactMtx ensures that no compaction gets triggered while
+	// changing the autoCompact var.
+	autoCompactMtx sync.Mutex
```

Comment:
Do we really need another mutex for this? Can we not use `cmtx` itself?

Reply:
Yeah I think blocking is the right way.
Imagine a processes that relies on the fact that after `DisableCompaction()`  no data changes will happen which won't be the case if we don't block. Otherwise if the process doesn't care than can always run `DisableCompaction()` in the background to continue the other executions. 
So in other words blocking leaves that decision to the caller which is always a good thing.